test/incrParse/simple.swift

```diff
@@ -13,15 +13,16 @@
 // RUN: %validate-incrparse %s --test-case LAST_CHARACTER_OF_STRUCT
 // RUN: %validate-incrparse %s --test-case ADD_ARRAY_CLOSE_BRACKET
 // RUN: %validate-incrparse %s --test-case ADD_IF_OPEN_BRACE
+// RUN: %validate-incrparse %s --test-case EXTEND_IDENTIFIER
 
 func start() {}
 
 <reparse REPLACE>
 func foo() {
 }
 
-_ = <<REPLACE<6|||7>>></reparse REPLACE>
-_ = <<REPLACE_BY_LONGER<6|||"Hello World">>>
+_ = <<REPLACE<6|||7>>>
+_ = </reparse REPLACE><<REPLACE_BY_LONGER<6|||"Hello World">>>
```

Comment:
Putting the `</reparse REPLACE>` before the `6` seemed really odd to me and it turns out there is another off-by-one error here, where we don't report the last unexpectedly reparsed character.

Investigating further, I noticed that we need to remove the `- 1` in `swift-syntax-test.cpp:252` plus its corresponding comment. I don't know if that's related to some of your changes on `SourceLoc` or has been a bug ever since, but it fixed the issue. After that the end repairs marker also needs to go at the end of the line.